graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/eacs/BEH7.kt

```diff
@@ -0,0 +1,241 @@
+package com.anaplan.engineering.azuki.graphs.eacs
+
+import com.anaplan.engineering.azuki.core.runner.Eac
+import com.anaplan.engineering.azuki.core.system.BEH
+import com.anaplan.engineering.azuki.graphs.adapter.api.GraphBehaviours
+import com.anaplan.engineering.azuki.graphs.dsl.GraphScenario
+import com.anaplan.engineering.azuki.graphs.graphA
+
+@BEH(GraphBehaviours.HasCycle, GraphBehaviours.CreateDirectedGraph, """
+    Determine the number of cycles in graph
+""")
+class BEH7 : GraphScenario() {
```

Comment:
At a glance, these scenarios look very similar to those in BEH6, so it's probably best I go through them thoroughly after you have addressed the comments there and made the corresponding changes here (if relevant).